app/ui/browser-modern/views/browser/page/index.jsx

```diff
@@ -69,28 +69,35 @@ class Page extends Component {
       }));
     });
 
-    this.webview.addEventListener('did-navigate', () => {
+    this.webview.addEventListener('did-navigate', e => {
       // Event fired when a page loads, after `did-start-loading`.
       // Only emitted once.
-      const url = this.webview.getURL();
-      const isTofino = url.startsWith(Endpoints.TOFINO_PROTOCOL);
-      this.props.dispatch(PageActions.setPageDetails(this.props.pageId, { location: url }));
-      this.props.dispatch(UIEffects.setURLBarValue(this.props.pageId, url));
-      this.props.dispatch(UIEffects.focusURLBar(this.props.pageId, { select: isTofino }));
+      const pageId = this.props.pageId;
+      const isTofino = e.url.startsWith(Endpoints.TOFINO_PROTOCOL);
+      const history = this.webview.getWebContents().history;
+      const historyIndex = this.webview.getWebContents().getActiveIndex();
```

Comment:
getWebContents() is a sync IPC call to the main process (sucks I know), so please cache the result rather than calling it twice.